UDP/course_materialized_view.md

```diff
@@ -22,8 +22,8 @@ on pe.person_id = cse.person_id join entity.person p on p.person_id = cse.person
 courses_enrollment as (select cst.*, e.person_id, e.uniqname, e.person_name from courses_sections_of_current_term cst join enrollment e on e.course_section_id = cst.course_section_id),
 course_grades as (select ce.*, cg.le_current_score as "Current_Grade", cg.le_final_score as "Final_Grade", cg.gpa_cumulative_excluding_course_grade as cumulative_gpa from entity.course_grade cg join courses_enrollment ce on ce.course_section_id = cg.course_section_id and ce.person_id = cg.person_id),
 
-avg_course_grade as (select "Canvas_Course_ID", round(AVG("Current_Grade"),2) as avg_course_grade  from course_grades group by  "Canvas_Course_ID"),
-courses_grade_average as (select cg.*, acg.avg_course_grade  from avg_course_grade acg join course_grades cg on cg."Canvas_Course_ID"  = acg."Canvas_Course_ID"),
+average_course_grade as (select "Canvas_Course_ID", round(AVG("Current_Grade"),2) as avg_course_grade from (select distinct uniqname, "Canvas_Course_ID", "Current_Grade" from course_grades) as course_unique_student_avg_grade group by "Canvas_Course_ID"),
```

Comment:
>Wait, I think this logic won't work with calculating averages, if student is in 2 different course section then I will be taking into consideration 2 grades of his. I will test and update the query.

This change should fix this. I checked with these courses (577751, 582106) which has students in multiple sections.

This course when student are in multiple academic program 585651

The logic I applied for calculating average after when pulling course grade and before academic major/program logic. This way the only multiple entries (rows) we have to worry is when student are in different sections in same course.

We can chat if you have any questions on this.

Reply:
@zqian If you want to test some point in subqueries sequentially, here is how you do it 

```
with
term_info as (select academic_term_id , name as term_name from entity.academic_term at2 where  current_timestamp < at2.term_end_date  
and current_timestamp > at2.term_begin_date ),
courses as 
(select co.course_offering_id, co.le_code as canvas_course_title, co.available_credits as credits, co.academic_term_id, term_name
from entity.course_offering co join term_info at2 on co.academic_term_id = at2.academic_term_id where co.le_status = 'available')  
select * from courses
```

This will only fetch result from until that point in subquery which is courses.

 then add `sections`  like below this will fetch both sections and course.

```
with
term_info as (select academic_term_id , name as term_name from entity.academic_term at2 where  current_timestamp < at2.term_end_date  
and current_timestamp > at2.term_begin_date ),
courses as 
(select co.course_offering_id, co.le_code as canvas_course_title, co.available_credits as credits, co.academic_term_id, term_name
from entity.course_offering co join term_info at2 on co.academic_term_id = at2.academic_term_id where co.le_status = 'available'),
sections as 
(select cs.course_section_id , cs2.lms_ext_id as canvas_section_id, cs.le_current_course_offering_id as course_offering_id, co4.lms_ext_id as "Canvas_Course_ID", cs.le_name as section_name, cs.section_number as "Section" from entity.course_section cs join keymap.course_section cs2 on cs.course_section_id = cs2.id join keymap.course_offering co4 on co4.id = cs.le_current_course_offering_id and cs.status='active'),
courses_sections_of_current_term as (select c.canvas_course_title, c.term_name, s.*, c.academic_term_id, c.credits from courses c join sections s on c.course_offering_id=s.course_offering_id)
select * from sections
```
This is one way to validate this is what I was doing when trying to calculate average course grade

Reply:
followed the comments and ran those queries.